### ISSUE TITLE
Store reconciliation timestamp to the Module CRD

### ISSUE THREAD

**Issue by petar-cvit** [cyclops-ui/cyclops#430]
**Is your feature request related to a problem? Please describe.**
There is currently [reconciliation status](https://github.com/cyclops-ui/cyclops/blob/cd0fdfda46bab2f4fcebf41bea8079b7168e4ff5/cyclops-ctrl/api/v1alpha1/module_types.go#L58) on the Module CRD, but another useful information to add to the status would be to add the time when the reconciliation finished. Also, the time should be then propagated to the [history entry](https://github.com/cyclops-ui/cyclops/blob/cd0fdfda46bab2f4fcebf41bea8079b7168e4ff5/cyclops-ctrl/api/v1alpha1/module_types.go#L91).

**Describe the solution you'd like**
As a solution, we should add a new field to the reconciliation status and [map it when setting the status](https://github.com/cyclops-ui/cyclops/blob/cd0fdfda46bab2f4fcebf41bea8079b7168e4ff5/cyclops-ctrl/internal/modulecontroller/module_controller.go#L413) of the module after reconciliation.

**Comment by SpiffyEight77** [comment]
Hey @petar-cvit, I'm interested in this issue. Could you please assign it to me? 🙇🏻‍♂️

**Comment by petar-cvit** [comment]
@SpiffyEight77 do you have any updates on the issue?

**Comment by SpiffyEight77** [comment]
@petar-cvit 
I apologize for my delayed response. I've been extremely busy with work and will be until October, so I don't think I'll have enough time to complete this task. I've removed myself from the assignment.

**Comment by Abiji-2020** [comment]
@petar-cvit  I could work on this issue 
Things needed to be done :
 - add a new field in the `ReconciliationStatus` struct for the HistoryEntry
 - Map the values of the status in the `setStatus` for the updated `ReconciliationStatus`

**Comment by petar-cvit** [comment]
@Abiji-2020 yup, that sounds good. Just make sure that after you update the struct in go, you generate the manifests using `make generate` and `make manifests` and apply created manifests to your cluster

**Comment by petar-cvit** [comment]
Hey @Abiji-2020, do you have any updates on the issue? Let me know if you have any questions

**Comment by Abiji-2020** [comment]
I have a doubt that currently before updating the code I tried to run the Makefile to generate but it is giving me nil pointer error and with no further details. So I am struck in that process

**Comment by petar-cvit** [comment]
Hey @Abiji-2020, thanks for bringing this up.

There is a mismatch between go and controller-gen versions. I bumped the controller-gen version in #540. Sync your fork and pull the latest main branch.

To install a new version and generate manifests, run the following commands from the `cyclops-ctrl` folder.

Remove previous controller-gen:
```
rm -rf bin
```
Install the new version:
```
make controller-gen
```
and then you can generate code (`make generate`) and manifests (`make manifests`).

**Comment by Abiji-2020** [comment]
oh okay, I was a beginner so I was wondering whether I am doing anything wrong here
- will update you after the correction
